app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt

```diff
@@ -267,34 +263,16 @@ class MainActivity : AppCompatActivity() {
     }
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-
-        val board = SystemInfoUtils.getSnpeSupportedBoard()
-
-        if (!SupportedSnapdragonBoards.isBoardSupported(board)) {
-            val text =
-                Html.fromHtml("The device is not supported, you need <b>Snapdragon-powered</b> device with <b>OpenCL</b> support, more details at <b>https://www.mapbox.com/android-docs/vision/overview/</b>")
-            Toast.makeText(this, text, Toast.LENGTH_LONG).show()
-            VisionLogger.e(
-                "NotSupportedBoard",
-                "Current board is {\"$board\"}, Supported Boards: [${enumValues<SupportedSnapdragonBoards>().joinToString { it.name }}]; System Info: [${SystemInfoUtils.obtainSystemInfo()}]"
-            )
-            finish()
-            return
-        }
 
         soundsPlayer = SoundsPlayer(this)
+    }
 
-        if (!allPermissionsGranted() && Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            requestPermissions(getRequiredPermissions(), PERMISSIONS_REQUEST_CODE)
-        } else {
-            onPermissionsGranted()
-        }
+    override fun initViews() {
+        setContentView(R.layout.activity_main)
     }
 
-    private fun onPermissionsGranted() {
+    public override fun onPermissionsGranted() {
```

Comment:
Is `public` needed here?

Reply:
yep, a base class is stored in a separate module